packages/token-bridge-sdk/src/hooks/useBalance.spec.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks'
+import { useBalance } from './useBalance'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+import { SWRConfig } from 'swr'
+import { PropsWithChildren } from 'react'
+
+// Create a new cache for every test
+const Container = ({ children }: PropsWithChildren<{}>) => (
+  <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
+)
+
+describe('useBalance', () => {
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  it('getter return null for undefined walletAddress', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+
+    const { result, waitForNextUpdate } = await renderHook(() =>
+      useBalance({
+        provider,
+        walletAddress: undefined
+      })
+    )
+
+    try {
+      await waitForNextUpdate({ timeout: 100 })
+    } catch (err) {}
+
+    const {
+      current: {
+        eth: [ethBalance]
+      }
+    } = result
+
+    expect(ethBalance).toBeNull()
+  })
+
+  it('getter return null for missing chainId', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL
+    )
+
+    // @ts-ignore
+    jest.spyOn(provider, 'getNetwork').mockImplementation(() => {
+      return null
+    })
```

Comment:
Same here on mocking `provider.getBalance`, too

Reply:
I wonder if there's a better way to mock this case, instead of returning `null` for `getNetwork`, because that should never be the case

Reply:
When useBalance was throwing an error, it was always because `network` was undefined, not because `chainId` was missing